### PR TITLE
add a function to remove row and column from LAPACKFullMatrix

### PR DESCRIPTION
**Pull request by davydden**
alternative names are:
* `remove_row_col`
* `delete_row_col`
* `delete()`

and probably other. 

The current choice is done in the hope to make the code readable `A.remove(3,5)` -> `remove row 3 and column 5.`

### CODE REVIEW

File: source/lac/lapack_full_matrix.cc [hunk 1]
```diff
@@ -88,13 +88,39 @@ LAPACKFullMatrix<number>::grow_or_shrink (const size_type n)
...
 
 
+template <typename number>
+void
+LAPACKFullMatrix<number>::remove (const size_type row, const size_type col)
+{
+  Assert (row<this->n_rows(), ExcIndexRange(row,0,this->n_rows()));
+  Assert (col<this->n_cols(), ExcIndexRange(col,0,this->n_cols()));
+
+  TransposeTable<number> copy(*this);
```
Comment (bangerth) [inline source/lac/lapack_full_matrix.cc]: Do we have move constructors? If so, maybe use
```
TransposeTable<number> copy(std::move(*this));
```
The memory is going to be released next anyway, so we might as well steal it.

**Comment by bangerth** [comment]
Maybe `remove_row_and_columns()`?

**Comment by davydden** [comment]
@bangerth I updated according to your comments.

**Comment by davydden** [comment]
./run-tests

**Comment by bangerth** [comment]
OK to merge once the tester is happy.

**Comment by davydden** [comment]
the failing test is the same unrelated
```
opencascade/arclength_boundary_02.debug (Failed)
```
so I take it as green light.

**Pull request merged**